receiver/macosunifiedloggingreceiver/receiver.go

```
@@ -34,20 +44,309 @@ func newUnifiedLoggingReceiver(
+	)
+	expBackoff.Reset()
+
+	ticker := backoff.NewTicker(expBackoff)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			count, err := r.runLogCommand(ctx, "")
+			if err != nil {
+				r.logger.Error("Failed to run log command", zap.Error(err))
+			}
+
+			// If logs were processed, reset backoff to minimum interval
+			if count > 0 {
+				r.logger.Debug("Logs found, resetting backoff to minimum interval", zap.Int("count", count))
+				expBackoff.Reset()
+			}
+			// If no logs, backoff will continue to increase to MaxPollInterval
+		}
+	}
+}
+
+// runLogCommand executes the log command and processes output
+// Returns the number of logs processed
+// archivePath should be empty string for live mode, or a specific archive path for archive mode
+//
+//nolint:unused // only used on darwin platform
+func (r *unifiedLoggingReceiver) runLogCommand(ctx context.Context, archivePath string) (int, error) {
+	// Build the log command arguments
+	args := r.buildLogCommandArgs(archivePath)
+
+	r.logger.Info("Running log command", zap.Strings("args", args))
+
+	// Create the command
+	cmd := exec.CommandContext(ctx, "log", args...) // #nosec G204 - args are controlled by config
+
+	// Get stdout pipe
+	stdout, err := cmd.StdoutPipe()
+	if err != nil {
+		return 0, fmt.Errorf("failed to get stdout pipe: %w", err)
+	}
+
+	// Start the command
+	if err := cmd.Start(); err != nil {
+		return 0, fmt.Errorf("failed to start log command: %w", err)
+	}
+
+	// Ensure the process is properly cleaned up to avoid zombies
+	defer func() {
+		_ = cmd.Wait()
+	}()
+
+	// Read and process output line by line
+	scanner := bufio.NewScanner(stdout)
+	// Set a large buffer size for long log lines
+	buf := make([]byte, 0, 1024*1024) // 1MB buffer
+	scanner.Buffer(buf, 10*1024*1024) // 10MB max
+
+	var processedCount int
+	isFirstLine := true
+	for scanner.Scan() {
+		select {
+		case <-ctx.Done():
+			_ = cmd.Process.Kill()
```

ignoring the error upon killing the process?